api/http/handler/registry.go

```diff
@@ -312,3 +332,44 @@ func (handler *RegistryHandler) handleDeleteRegistry(w http.ResponseWriter, r *h
 		return
 	}
 }
+
+// validateRegistryURL validates wether given url is valid a docker registry url by
+// sending http get request to url using combination of protocols and available
+// registry versions. upon first successfull attempt, it returns protocol, version
+// and nil error.
+func validateRegistryURL(url string) (string, string, error) {
+	if !strings.HasSuffix(url, "/") {
```

Comment:
Is this actually required? What happens if we always add the extra slash? I believe that `my.registry.com//v1` would be valid for example.